.github/workflows/Image.yaml

```diff
@@ -459,34 +469,14 @@ jobs:
           path: ${{ needs.prepare-build.outputs.oci-img-path }}/image.yaml
           key: ${{ github.run_id }}-image-trigger
 
-      - name: Dispatch Releases workflow
-        # Using this action because others can have this problem:
-        # https://github.com/convictional/trigger-workflow-and-wait/issues/61
-        uses: mathze/workflow-dispatch-action@v1.2.0
-        id: run-releases
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          ref: ${{ github.ref_name }}
-          fail-on-error: true
-          workflow-name: Release.yaml
-          payload: '{ "oci-image-name": "${{ needs.prepare-build.outputs.oci-img-name }}", "image-trigger-cache-key": "${{ github.run_id }}-image-trigger"}'
-          run-id: dummy
-          trigger-timeout: "10m"
-          wait-timeout: "20m"
-          use-marker-step: true
-
-      - name: Write step summary
-        run: |
-          url='${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ steps.run-releases.outputs.run-id }}'
-          echo " - Triggered releases for '${{ needs.prepare-build.outputs.oci-img-name }}' at [${url}](${url})" >> "$GITHUB_STEP_SUMMARY"
-
-      - name: Enforce release conclusion
-        if: ${{ steps.run-releases.outputs.run-conclusion != 'success' }}
-        # The previous step doesn't always raise an error
-        run: |
-          url='${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ steps.run-releases.outputs.run-id }}'
-          echo "Failed to release '${{ needs.prepare-build.outputs.oci-img-name }}' at [${url}](${url})."
-          exit 1
+  release:
+    name: Release
+    needs: [prepare-releases]
+    uses: ./.github/workflows/Release.yaml
+    with:
+      oci-image-name: "${{ inputs.oci-image-name }}"
```

Comment:
This will only work when Image.yaml is called, and not on `push` or other events.

Reply:
@cjdcordeiro It was `${{ needs.prepare-build.outputs.oci-img-name }}` before, but it was strangely giving an empty value, which is very peculiar since previous jobs are able to read that variable value without issue.

Reply:
@linostar why do you say it is empty? It is being set [here](https://github.com/canonical/oci-factory/actions/runs/7875730443/job/21488246913#step:5:29) (in the previously failed job)

Reply:
it seems it was not being passed to the called workflow... investigating

Reply:
Ok I think I might know what it is.

In https://github.com/canonical/oci-factory/blob/d939c480de447ece234fdd15f9b914df36264ba9/.github/workflows/Image.yaml#L76, the inference of that variable is based on whether the workflow has been dispatched or not.

Well, now that you have introduced `workflow_call`, that `if` will be executed (cause the workflow is being **called** and not **disptached**) and that seems to be resulting in those variables being wrongfully set to nothing. 

You need to expand that `if` condition to also encompass the `workflow_call` event

Reply:
Should work now:
https://github.com/canonical/oci-factory/actions/runs/7888207148/job/21525424565

I think it was also missing `prepare-build` in `needs` list before, because otherwise it is not able to access `needs.prepare-build.outputs`.

Reply:
And it didn't work. Back to investigating...

Reply:
it was an unrelated LP error